### PR TITLE
refactor(cloud): unify input key/value resource reading and mapping

### PR DESCRIPTION
**Pull request by vvagaytsev**
**What this PR does / why we need it**:

Align the flow with the `users create` command. Stick to the same validation order.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: core/src/commands/cloud/secrets/secrets-create.ts [hunk 1]
```diff
@@ -86,20 +87,20 @@ export class SecretsCreateCommand extends Command<Args, Opts> {
...
+      resourceName: "secret",
+      log: cmdLog,
+    })
 
     const api = garden.cloudApi
     if (!api) {
       throw new ConfigurationError({ message: noApiMsg("create", "secrets") })
     }
 
+    const secretsToCreate: Secret[] = inputSecrets.map(([key, value]): Secret => ({ name: key, value }))
```
Comment (stefreak) [inline core/src/commands/cloud/secrets/secrets-create.ts:102]: The outer type will be inferred automatically, if you want the code to be less verbose. But it's fine for me like this too 👍 
```suggestion
    const secretsToCreate = inputSecrets.map(([key, value]): Secret => ({ name: key, value }))
```

**Review by stefreak — APPROVED** [review]
👍 Thanks, makes sense!

File: core/src/commands/cloud/secrets/secrets-create.ts (hunk 1 above)
Comment (vvagaytsev) [reply core/src/commands/cloud/secrets/secrets-create.ts:102]: I left it intentionally here and in the rest 2 places. Just to ensure the type-safety :) Is for some reason the inner type is removed, the outer guard will still fail if the mapper returns an incompatible value :D

Comment (stefreak) [reply core/src/commands/cloud/secrets/secrets-create.ts:102]: A little excessive for my taste to repeat the type in the same one-liner, but both variants are acceptable for me :)

**Pull request merged**